### PR TITLE
[SYCL-MLIR] Enable raise to affine by default

### PR DESCRIPTION
**Pull request by whitneywhtsang**
We want to enable `-raise-scf-to-affine` by default, so we can run affine optimizations.

Signed-off-by: Tsang, Whitney <whitney.tsang@intel.com>

### CODE REVIEW

**Review by etiotto — APPROVED** [review]

File: polygeist/tools/cgeist/Test/Verification/arrayconsllvm.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: cgeist %s --function=* -S | FileCheck %s
+// RUN: cgeist %s --function=* -S --raise-scf-to-affine=false | FileCheck %s
```
Comment (sommerlukas) [inline polygeist/tools/cgeist/Test/Verification/arrayconsllvm.cpp:1]: What's the reason to deactivate the pass for this test and the other tests changed in this PR? 

Does the pass break the test?

Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Test/Verification/arrayconsllvm.cpp:1]: Yes, take this test as an example, the CHECK for the loop is no longer `scf.for`.
As these tests are not written by us, I think is best to keep the original CHECKs.

Comment (sommerlukas) [reply polygeist/tools/cgeist/Test/Verification/arrayconsllvm.cpp:1]: Ok, I see. 

We might get more test-coverage for the raising pass if we also enabled it for non-SYCL tests, but that's probably something we can look into at another time.

**Review by sommerlukas — APPROVED** [review]
LGTM.

**Pull request merged**